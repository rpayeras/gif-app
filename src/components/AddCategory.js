import React, { useState } from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ({onSubmit}) => {
    const [inputText, setInputText] = useState('')

    const handleInput = (e) => setInputText(e.target.value);

    const handleSubmit = (e) => {
        e.preventDefault()

        if(inputText.trim().length > 2){
            onSubmit(inputText)
            setInputText('')
        }
    }

    return (
        <>
            <form onSubmit={handleSubmit}>
                <input name="name" type="text" value={inputText} onChange={handleInput} placeholder="Add a category..."/>
            </form>
        </>
    );
};

AddCategory.propTypes = {
    onSubmit: PropTypes.func.isRequired
}