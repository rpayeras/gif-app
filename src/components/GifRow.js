import React, { useState, useEffect } from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifRow = ({category}) => {
  const {data: images, loading} = useFetchGifs(category)

  return (
    <>
      <h3>{category}</h3>
      {loading ?? <p>'Loading...'</p>}
      { <div className='card-grid '>
        {images.map((img) => {
          return (
            <GifGridItem {...img} key={img.id} />
          )
        })}
    </div> }
    </>
  );
};

GifRow.propTypes = {
  category: PropTypes.string.isRequired
}