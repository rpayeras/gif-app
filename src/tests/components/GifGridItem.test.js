import React from 'react';
import { shallow } from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';


describe('Testing GifGridItem component', () => {
    const img = {
        title: 'Alonso',
        url: 'https://www.google.es'
    }
    const wrapper = shallow(<GifGridItem {...img}/>)

    test('Should show component', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Should show a title', () => {
        expect(wrapper.find('h4').text().trim()).toBe(img.title)
    })

    test('Should show an image', () => {
        expect(wrapper.find('img').prop('src')).toBe(img.url)
    })

    test('Should have animate fadeIn', () => {
        expect(wrapper.find('div').prop('className').includes('animate__fadeIn')).toBe(true)
    })
})