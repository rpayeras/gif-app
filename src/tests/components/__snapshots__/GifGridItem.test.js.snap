// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem component Should show component 1`] = `
<div
  className="card animate__animated  animate__fadeIn"
>
  <img
    alt="Alonso"
    src="https://www.google.es"
  />
  <h4>
    Alonso
  </h4>
</div>
`;
